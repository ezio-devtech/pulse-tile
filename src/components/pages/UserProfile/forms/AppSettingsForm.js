import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';

@reduxForm({ form: 'appSettingsFormSelector' })
export default class AppSettingsForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="appSettingsForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Application Title"
                      name="applicationTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Application Logo File"
                      name="logoPath"
                      component={FileInput}
                      id="logoPath"
                    />
                    <Field
                      label="Application Themes"
                      name="green"
                      component={Select}
                      options={optionsForThemesField}
                    />
                    <Field
                      label="Browser Window title"
                      name="browserWindowTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
