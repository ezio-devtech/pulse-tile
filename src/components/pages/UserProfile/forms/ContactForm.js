import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from './validation';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Address"
                      name="address"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="City"
                      name="city"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="State"
                      name="state"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Postal Code"
                      name="postalCode"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Country"
                      name="country"
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Phone Number"
                      name="phone"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Email"
                      name="email"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
