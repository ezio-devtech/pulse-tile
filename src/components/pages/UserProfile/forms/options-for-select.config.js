const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

export { optionsForThemesField, optionsForGenderField }
