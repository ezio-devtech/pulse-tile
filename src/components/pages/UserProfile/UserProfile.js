import React, { PureComponent } from 'react';
import { Row, Col } from 'react-bootstrap';
import classNames from 'classnames';
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux';
import _ from 'lodash/fp';
import moment from 'moment';

import PersonalInformationPanel from './PersonalInformationPanel';
import AppSettingsForm from './forms/AppSettingsForm';
import PersonalForm from './forms/PersonalForm';
import ContactForm from './forms/ContactForm';
import { formStateSelector, patientInfoSelector, userAccountSelector } from './selectors';
import { fetchProfileAppPreferencesRequest } from '../../../ducks/fetch-profile-application-preferences.duck';
import { fetchPatientsInfoRequest } from '../../../ducks/fetch-patients-info.duck';
import { setLogo } from '../../../ducks/set-logo.duck';
import { setTitle } from '../../../ducks/set-title.duck';
import { setTheme } from '../../../ducks/set-theme.duck';
import { valuesSettingsFormLabels, valuesPersonalFormLabels, valuesContactFormLabels } from './forms/values-names.config';
import themes from './theme-config';

const APPLICATION_PREFERENCES = 'applicationPreferences';
const PERSONAL_INFORMATION = 'personalInformation';
const CONTACT_INFORMATION = 'contactInformation';
const CHANGE_HISTORY = 'changeHistory';

const mapDispatchToProps = dispatch => ({ actions: bindActionCreators({ fetchProfileAppPreferencesRequest, fetchPatientsInfoRequest, setLogo, setTitle, setTheme }, dispatch) });

@connect(formStateSelector)
@connect(userAccountSelector)
@connect(patientInfoSelector, mapDispatchToProps)
class UserProfile extends PureComponent {
  state = {
    openedPanel: APPLICATION_PREFERENCES,
    expandedPanel: 'all',
    isAllPanelsVisible: false,
    editedPanel: {},
  };

  handleShow = (name) => {
    this.setState({ openedPanel: name })
  };

  handleExpand = (name) => {
    if (this.state.expandedPanel === 'all') {
      this.setState(prevState => ({ expandedPanel: name, openedPanel: name, isAllPanelsVisible: !prevState.isAllPanelsVisible }));
    } else {
      this.setState(prevState => ({ expandedPanel: 'all', isAllPanelsVisible: !prevState.isAllPanelsVisible }));
    }
  };

  handleEdit = (name) => {
    this.setState(prevState => ({
      editedPanel: {
        ...prevState.editedPanel,
        [name]: true,
      },
    }))
  };

  handleCancel = (name) => {
    const { patientsInfo, dispatch } = this.props;
    dispatch(setTheme(patientsInfo.themeColor));
    this.setState(prevState => ({
      editedPanel: {
        ...prevState.editedPanel,
        [name]: false,
      },
    }))
  };

  handleSaveSettingsForm = (formValues, name) => {
    const { actions, formState, patientsInfo, dispatch } = this.props;
    Object.keys(patientsInfo).forEach((key) => {
      patientsInfo[key] = formValues[key];
    });
    if (_.isEmpty(formState.syncErrors.title) && _.isEmpty(formState.syncErrors.browserTitle)) {
      dispatch(setLogo(patientsInfo.logoB64));
      dispatch(setTitle(patientsInfo.browserTitle));
      actions.fetchProfileAppPreferencesRequest(formValues);
      this.setState(prevState => ({
        editedPanel: {
          ...prevState.editedPanel,
          [name]: false,
        },
      }))
    }
  };

  render() {
    const { openedPanel, expandedPanel, isAllPanelsVisible, editedPanel } = this.state;
    const { formState, patientsInfo, user } = this.props;

    const theme = themes[patientsInfo.themeColor] ? themes[patientsInfo.themeColor] : themes.default;
    const currentDate = (new Date()).getTime();
    const CONVERT_CURRENT_DATE = moment(currentDate).format('DD-MMM-YYYY');
    const CONVERT_CURRENT_DATE_WITH_TIME = moment(currentDate).format('YYYY-MM-DD HH:mm');

    return (<section className="page-wrapper">
      <div className={classNames('section', { 'full-panel full-panel-main': isAllPanelsVisible })}>
        <Row>
          <Col xs={12}>
            <div className="section-main">
              <div className="panel-group accordion">
                {(expandedPanel === 'applicationPreferences' || expandedPanel === 'all') && !editedPanel[APPLICATION_PREFERENCES] ? <PersonalInformationPanel
                  name={APPLICATION_PREFERENCES}
                  title="Application Preferences"
                  isOpen={openedPanel === APPLICATION_PREFERENCES}
                  onShow={this.handleShow}
                  onExpand={this.handleExpand}
                  onEdit={this.handleEdit}
                  editedPanel={editedPanel}
                  onCancel={this.handleCancel}
                >
                  <div className="panel-body-inner">
                    <div className="form">
                      <div className="form-group-wrapper">
                        <Row>
                          <Col xs={12} md={6}>
                            <Row>
                              <div className="col-md-11">
                                <div className="form-group">
                                  <label className="control-label">{valuesSettingsFormLabels.APP_TITLE}</label>
                                  <div className="form-control-static">{patientsInfo.title}</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesSettingsFormLabels.LOGO_PATH}</label>
                                  <div className="form-control-static">
                                    <img src={patientsInfo.logoB64} alt="Logo Example" />
                                  </div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesSettingsFormLabels.SELECT_THEME_ONE}</label>
                                  <div className="palette-color">
                                    <span className="palette-color-icon" style={{ background: theme.baseColor }}></span>
                                    <span className="palette-color-name">{theme.name}</span>
                                  </div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesSettingsFormLabels.BROWSER_TITLE}</label>
                                  <div className="form-control-static">{patientsInfo.browserTitle}</div>
                                </div>
                              </div>
                            </Row>
                          </Col>
                        </Row>
                      </div>
                    </div>
                  </div>
                </PersonalInformationPanel> : null }
                {(expandedPanel === 'applicationPreferences' || expandedPanel === 'all') && editedPanel[APPLICATION_PREFERENCES] ? <PersonalInformationPanel
                  name={APPLICATION_PREFERENCES}
                  title="Application Preferences"
                  isOpen={openedPanel === APPLICATION_PREFERENCES}
                  onShow={this.handleShow}
                  onExpand={this.handleExpand}
                  onEdit={this.handleEdit}
                  editedPanel={editedPanel}
                  onCancel={this.handleCancel}
                  onSaveSettings={this.handleSaveSettingsForm}
                  formValues={formState.values}
                >
                  <AppSettingsForm
                    patientsInfo={patientsInfo}
                  />
                </PersonalInformationPanel> : null }

                {(expandedPanel === 'personalInformation' || expandedPanel === 'all') && !editedPanel[PERSONAL_INFORMATION] ? <PersonalInformationPanel
                  name={PERSONAL_INFORMATION}
                  title="Personal Information"
                  isOpen={openedPanel === PERSONAL_INFORMATION}
                  onShow={this.handleShow}
                  onExpand={this.handleExpand}
                  onEdit={this.handleEdit}
                  editedPanel={editedPanel}
                  onCancel={this.handleCancel}
                >
                  <div className="panel-body-inner">
                    <div className="form">
                      <div className="form-group-wrapper">
                        <Row>
                          <Col xs={12} md={6}>
                            <Row>
                              <Col md={11}>
                                <div className="form-group">
                                  <label className="control-label">{valuesPersonalFormLabels.FIRST_NAME}</label>
                                  <div className="form-control-static">{user.given_name}</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesPersonalFormLabels.LAST_NAME}</label>
                                  <div className="form-control-static">{user.family_name}</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesPersonalFormLabels.NHS_NUMBER}</label>
                                  {user.role === 'IDCR'
                                    ? <div className="form-control-static" />
                                    : <div className="form-control-static">{user.nhsNumber}</div> }
                                </div>
                              </Col>
                            </Row>
                          </Col>
                          <Col xs={12} md={6}>
                            <Row>
                              <div className="col-md-11 col-md-offset-1">
                                <div className="form-group">
                                  <label className="control-label">{valuesPersonalFormLabels.DATE_OF_BIRTH}</label>
                                  <div className="form-control-static ng-binding">{CONVERT_CURRENT_DATE}</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesPersonalFormLabels.SELECT_GENDER}</label>
                                  <div className="form-control-static ng-binding">Female</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesPersonalFormLabels.DOCTOR}</label>
                                  <div className="form-control-static ng-binding">Dr Emma Huston</div>
                                </div>
                              </div>
                            </Row>
                          </Col>
                        </Row>
                      </div>
                    </div>
                  </div>
                </PersonalInformationPanel> : null }
                {(expandedPanel === 'personalInformation' || expandedPanel === 'all') && editedPanel[PERSONAL_INFORMATION] ? <PersonalInformationPanel
                  name={PERSONAL_INFORMATION}
                  title="Personal Information"
                  isOpen={openedPanel === PERSONAL_INFORMATION}
                  onShow={this.handleShow}
                  onExpand={this.handleExpand}
                  onEdit={this.handleEdit}
                  editedPanel={editedPanel}
                  onCancel={this.handleCancel}
                  onClick={this.handleClick}
                >
                  <PersonalForm />
                </PersonalInformationPanel> : null }

                {(expandedPanel === 'contactInformation' || expandedPanel === 'all') && !editedPanel[CONTACT_INFORMATION] ? <PersonalInformationPanel
                  name={CONTACT_INFORMATION}
                  title="Contact Information"
                  isOpen={openedPanel === CONTACT_INFORMATION}
                  onShow={this.handleShow}
                  onExpand={this.handleExpand}
                  onEdit={this.handleEdit}
                  editedPanel={editedPanel}
                  onCancel={this.handleCancel}
                >
                  <div className="panel-body-inner">
                    <div className="form">
                      <div className="form-group-wrapper">
                        <div className="row">
                          <div className="col-xs-12 col-md-6">
                            <div className="row">
                              <div className="col-md-11">
                                <div className="form-group">
                                  <label className="control-label">{valuesContactFormLabels.ADDRESS}</label>
                                  <div className="form-control-static ng-binding">6801 Tellus Street</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesContactFormLabels.CITY}</label>
                                  <div className="form-control-static ng-binding">Westmorland</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesContactFormLabels.STATE}</label>
                                  <div className="form-control-static ng-binding">Westmorland</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesContactFormLabels.POSTAL_CODE}</label>
                                  <div className="form-control-static ng-binding">Box 306</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesContactFormLabels.SELECT_COUNTRY}</label>
                                  <div className="form-control-static ng-binding">USA</div>
                                </div>
                              </div>
                            </div>
                          </div>
                          <div className="col-xs-12 col-md-6">
                            <div className="row">
                              <div className="col-md-11 col-md-offset-1">
                                <div className="form-group">
                                  <label className="control-label">{valuesContactFormLabels.PHONE}</label>
                                  <div className="form-control-static ng-binding">07624 647524</div>
                                </div>

                                <div className="form-group">
                                  <label className="control-label">{valuesContactFormLabels.EMAIL}</label>
                                  <div className="form-control-static">{user.email}</div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </PersonalInformationPanel> : null }
                {(expandedPanel === 'contactInformation' || expandedPanel === 'all') && editedPanel[CONTACT_INFORMATION] ? <PersonalInformationPanel
                  name={CONTACT_INFORMATION}
                  title="Contact Information"
                  isOpen={openedPanel === CONTACT_INFORMATION}
                  onShow={this.handleShow}
                  onExpand={this.handleExpand}
                  onEdit={this.handleEdit}
                  editedPanel={editedPanel}
                  onCancel={this.handleCancel}
                  onClick={this.handleClick}
                >
                  <ContactForm />
                </PersonalInformationPanel> : null }

                {expandedPanel === 'changeHistory' || expandedPanel === 'all' ? <PersonalInformationPanel
                  name={CHANGE_HISTORY}
                  title="Change History"
                  isOpen={openedPanel === CHANGE_HISTORY}
                  onShow={this.handleShow}
                  onExpand={this.handleExpand}
                  onEdit={this.handleEdit}
                  editedPanel={editedPanel}
                  onCancel={this.handleCancel}
                >
                  <div className="panel-body-inner">
                    <div className="form">
                      <div className="form-group-wrapper">
                        <div className="form-group">
                          <label className="control-label ng-binding">Change #1 Date</label>
                          <div className="form-control-static ng-binding">{CONVERT_CURRENT_DATE_WITH_TIME}</div>
                        </div>
                        <div className="form-group">
                          <label className="control-label">Changes</label>
                          <div className="form-control-static ng-binding">Last Name: <em className="ng-binding">White</em> <span className="next-separate"><i className="fa fa-caret-right" /></span> Blackwell</div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div className="panel-body-inner">
                    <div className="form">
                      <div className="form-group-wrapper">
                        <div className="form-group">
                          <label className="control-label ng-binding">Change #2 Date</label>
                          <div className="form-control-static ng-binding">{CONVERT_CURRENT_DATE_WITH_TIME}</div>
                        </div>
                        <div className="form-group">
                          <label className="control-label">Changes</label>
                          <div className="form-control-static ng-binding">Address: <em className="ng-binding">Flower Street</em> <span className="next-separate"><i className="fa fa-caret-right" /></span> 6801 Tellus Street</div>
                        </div>
                      </div>
                    </div>
                  </div>
                </PersonalInformationPanel> : null }
              </div>
            </div>
          </Col>
        </Row>
      </div>
    </section>
    )
  }
}

export default UserProfile

