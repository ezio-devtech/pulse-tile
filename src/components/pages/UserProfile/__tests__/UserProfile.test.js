import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';
import moment from 'moment';

Enzyme.configure({ adapter: new Adapter() });

import UserProfile from '../UserProfile';
import { valuesSettingsFormLabels, valuesPersonalFormLabels, valuesContactFormLabels } from '../forms.config';
import themes from '../theme-config';

const patientsInfo = {
  title: 'ripple',
  logoB64: 'data:image/gif;base64,R0lGODlhDwAPAKECAAAAzMzM/////wAAACwAAAAADwAPAAACIISPeQHsrZ5ModrLlN48CXF8m2iQ3YmmKqVlRtW4MLwWACH+H09wdGltaXplZCBieSBVbGVhZCBTbWFydFNhdmVyIQAAOw==',
  themeColor: 'green',
  browserTitle: 'PulseTile',
};
const patientsInfoForPurpleTheme = {
  title: 'ripple',
  logoB64: '',
  themeColor: 'purple',
  browserTitle: 'PulseTile',
};
const userAccount = {
  sub: '28AD8576-1948-4C84-8B5E-55FB7EE027CE',
  given_name: 'Bob',
  family_name: 'Smith',
  email: 'bob.smith@gmail.com',
  tenant: null,
  role: 'IDCR',
  roles: [
    'IDCR',
  ],
};

const userAccountPHR = {
  sub: 'auth0|59dcd0cc6311a61a835a2772',
  given_name: 'User',
  family_name: 'Test1',
  email: 'suvenir9502@gmail.com',
  nhsNumber: '9999999000',
  tenant: null,
  role: 'PHR',
  roles: [
    'PHR',
  ],
};

const mockStore = configureStore();
const store = mockStore({ patientsInfo, userAccount });
const storeForPurpleTheme = mockStore({ patientsInfo: patientsInfoForPurpleTheme, userAccount });
const storePHR = mockStore({ patientsInfo, userAccount: userAccountPHR });

const DATE_TO_USE = new Date('2017');
const DATE_TO_USE_TIME = DATE_TO_USE.getTime();
global.Date = jest.fn(() => DATE_TO_USE);
Date.now = jest.fn(() => DATE_TO_USE_TIME);

const CONVERT_CURRENT_DATE = moment(DATE_TO_USE_TIME).format('DD-MMM-YYYY');
const CONVERT_CURRENT_DATE_WITH_TIME = moment(DATE_TO_USE_TIME).format('YYYY-MM-DD HH:mm');

const match = {
  params: {},
  path: '/profile',
  url: '/profile',
};

const context = {
  router: {
    history: {
      push: () => {},
      replace: () => {},
      location: {
        pathname: '/profile',
      },
    },
    match,
    patientsInfo,
  },
};

describe('Component <UserProfile />', () => {
  it('should renders with all props correctly', () => {
    const component = shallow(
      <UserProfile
        store={store}
        match={match}
      />, { context }).dive().dive().dive();

    expect(component).toMatchSnapshot();

    expect(component.find('ControlPanel')).toHaveLength(4);
    expect(component.find('ReduxForm')).toHaveLength(0);

    const panelApp = component.find('ControlPanel').at(0);
    const panelPersonal = component.find('ControlPanel').at(1);
    const panelContact = component.find('ControlPanel').at(2);
    const panelHistory = component.find('ControlPanel').at(3);
    const colorName = themes[patientsInfo.themeColor] ? themes[patientsInfo.themeColor].name : themes.default.name;

    expect(panelApp.find('.control-label').at(0).text()).toEqual(valuesSettingsFormLabels.APP_TITLE);
    expect(panelApp.find('.control-label').at(1).text()).toEqual(valuesSettingsFormLabels.LOGO_PATH);
    expect(panelApp.find('.control-label').at(2).text()).toEqual(valuesSettingsFormLabels.SELECT_THEME_ONE);
    expect(panelApp.find('.control-label').at(3).text()).toEqual(valuesSettingsFormLabels.BROWSER_TITLE);

    expect(panelApp.find('.form-control-static').at(0).text()).toEqual(patientsInfo.title);
    expect(panelApp.find('img').props().src).toEqual(patientsInfo.logoB64);
    expect(panelApp.find('.palette-color-name').text()).toEqual(colorName);
    expect(panelApp.find('.form-control-static').at(2).text()).toEqual(patientsInfo.browserTitle);

    expect(panelPersonal.find('.control-label').at(0).text()).toEqual(valuesPersonalFormLabels.FIRST_NAME);
    expect(panelPersonal.find('.control-label').at(1).text()).toEqual(valuesPersonalFormLabels.LAST_NAME);
    expect(panelPersonal.find('.control-label').at(2).text()).toEqual(valuesPersonalFormLabels.NHS_NUMBER);
    expect(panelPersonal.find('.control-label').at(3).text()).toEqual(valuesPersonalFormLabels.DATE_OF_BIRTH);
    expect(panelPersonal.find('.control-label').at(4).text()).toEqual(valuesPersonalFormLabels.SELECT_GENDER);
    expect(panelPersonal.find('.control-label').at(5).text()).toEqual(valuesPersonalFormLabels.DOCTOR);

    expect(panelContact.find('.control-label').at(0).text()).toEqual(valuesContactFormLabels.ADDRESS);
    expect(panelContact.find('.control-label').at(1).text()).toEqual(valuesContactFormLabels.CITY);
    expect(panelContact.find('.control-label').at(2).text()).toEqual(valuesContactFormLabels.STATE);
    expect(panelContact.find('.control-label').at(3).text()).toEqual(valuesContactFormLabels.POSTAL_CODE);
    expect(panelContact.find('.control-label').at(4).text()).toEqual(valuesContactFormLabels.SELECT_COUNTRY);
    expect(panelContact.find('.control-label').at(5).text()).toEqual(valuesContactFormLabels.PHONE);
    expect(panelContact.find('.control-label').at(6).text()).toEqual(valuesContactFormLabels.EMAIL);

    expect(panelHistory.find('.panel-body-inner')).toHaveLength(2);
    expect(panelHistory.find('.form-group')).toHaveLength(4);
  });

  it('should renders correctly when open Edit Panel of Application Preferences', () => {
    const component = shallow(
      <UserProfile
        store={store}
        match={match}
      />, { context }).dive().dive().dive();
    component.setState({ editedPanel: { applicationPreferences: true } });

    expect(component).toMatchSnapshot();

    expect(component.find('ControlPanel')).toHaveLength(4);
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('ControlPanel').at(0).find('ReduxForm')).toHaveLength(1);
  });

  it('should renders correctly when open Edit Panel of Personal Information', () => {
    const component = shallow(
      <UserProfile
        store={store}
        match={match}
      />, { context }).dive().dive().dive();
    component.setState({ editedPanel: { personalInformation: true } });

    expect(component).toMatchSnapshot();

    expect(component.find('ControlPanel')).toHaveLength(4);
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('ControlPanel').at(1).find('ReduxForm')).toHaveLength(1);
  });

  it('should renders correctly when open Edit Panel of Contact Information', () => {
    const component = shallow(
      <UserProfile
        store={store}
        match={match}
      />, { context }).dive().dive().dive();
    component.setState({ editedPanel: { contactInformation: true } });

    expect(component).toMatchSnapshot();

    expect(component.find('ControlPanel')).toHaveLength(4);
    expect(component.find('ReduxForm')).toHaveLength(1);
    expect(component.find('ControlPanel').at(2).find('ReduxForm')).toHaveLength(1);
  });

  it('should work all methods of component', () => {
    const component = shallow(
      <UserProfile
        store={store}
        match={match}
      />, { context }).dive().dive().dive();

    component.instance().handleShow('applicationPreferences');
    expect(component.state().openedPanel).toEqual('applicationPreferences');
    component.instance().handleShow('personalInformation');
    expect(component.state().openedPanel).toEqual('personalInformation');

    expect(component.state().expandedPanel).toEqual('all');
    component.instance().handleExpand('applicationPreferences');
    expect(component.state().expandedPanel).toEqual('applicationPreferences');
    component.instance().handleExpand('applicationPreferences');
    expect(component.state().expandedPanel).toEqual('all');

    expect(!!component.state().editedPanel.applicationPreferences).toEqual(false);
    component.instance().handleEdit('applicationPreferences');
    expect(component.state().editedPanel.applicationPreferences).toEqual(true);
    component.instance().handleCancel('applicationPreferences');
    expect(component.state().editedPanel.applicationPreferences).toEqual(false);

    component.setProps({ formState: { syncErrors: { title: 'title' } } });
    component.instance().handleSaveSettingsForm(patientsInfo, 'applicationPreferences');
    component.setProps({ formState: { syncErrors: { browserTitle: 'title' } } });
    component.instance().handleSaveSettingsForm(patientsInfo, 'applicationPreferences');
    component.setProps({ formState: { syncErrors: {} } });
    component.instance().handleSaveSettingsForm(patientsInfo, 'applicationPreferences');
  });

  it('should renders correctly when theme is Purple', () => {
    const component = shallow(
      <UserProfile
        store={storeForPurpleTheme}
        match={match}
      />, { context }).dive().dive().dive();

    expect(component.find('ControlPanel').at(0).find('.palette-color-name').text()).toEqual(themes.purple.name);
  });

  it('should renders correctly when the User is PHR', () => {
    const component = shallow(
      <UserProfile
        store={storePHR}
        match={match}
      />, { context }).dive().dive().dive();

    expect(component.find('ControlPanel')).toHaveLength(4);
    expect(component.find('ReduxForm')).toHaveLength(0);

    const panelPersonal = component.find('ControlPanel').at(1);
    const panelContact = component.find('ControlPanel').at(2);
    const panelHistory = component.find('ControlPanel').at(3);

    expect(panelPersonal.find('.form-control-static').at(0).text()).toEqual(userAccountPHR.given_name);
    expect(panelPersonal.find('.form-control-static').at(1).text()).toEqual(userAccountPHR.family_name);
    expect(panelPersonal.find('.form-control-static').at(2).text()).toEqual(userAccountPHR.nhsNumber);
    expect(panelPersonal.find('.form-control-static').at(3).text()).toEqual(CONVERT_CURRENT_DATE);

    expect(panelContact.find('.form-control-static').at(6).text()).toEqual(userAccountPHR.email);

    expect(panelHistory.find('.form-control-static').at(0).text()).toEqual(CONVERT_CURRENT_DATE_WITH_TIME);
    expect(panelHistory.find('.form-control-static').at(2).text()).toEqual(CONVERT_CURRENT_DATE_WITH_TIME);

    expect(component).toMatchSnapshot();
  });
});
