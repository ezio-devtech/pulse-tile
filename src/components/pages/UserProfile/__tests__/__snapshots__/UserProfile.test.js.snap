// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <UserProfile /> should renders correctly when theme is Purple 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="panel-group accordion"
  >
    <ApplicationPreferencesPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      formState={Object {}}
      isSaveButton={true}
      isShowControlPanel={true}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onSaveSettings={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
      patientsInfo={
        Object {
          "browserTitle": "PulseTile",
          "logoB64": "",
          "themeColor": "purple",
          "title": "ripple",
        }
      }
      theme={
        Object {
          "baseColor": "#461a5a",
          "name": "Purple Theme",
        }
      }
    />
    <PersonalInformationPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      isSaveButton={false}
      isShowControlPanel={true}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
      user={
        Object {
          "email": "bob.smith@gmail.com",
          "family_name": "Smith",
          "given_name": "Bob",
          "role": "IDCR",
          "roles": Array [
            "IDCR",
          ],
          "sub": "28AD8576-1948-4C84-8B5E-55FB7EE027CE",
          "tenant": null,
        }
      }
    />
    <ContactInformationPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      isSaveButton={false}
      isShowControlPanel={true}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
      user={
        Object {
          "email": "bob.smith@gmail.com",
          "family_name": "Smith",
          "given_name": "Bob",
          "role": "IDCR",
          "roles": Array [
            "IDCR",
          ],
          "sub": "28AD8576-1948-4C84-8B5E-55FB7EE027CE",
          "tenant": null,
        }
      }
    />
    <ChangeHistoryPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      isSaveButton={false}
      isShowControlPanel={false}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
    />
    <FeedsPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      isSaveButton={false}
      isShowControlPanel={false}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
    />
  </div>
</section>
`;

exports[`Component <UserProfile /> should renders with all props correctly 1`] = `
<section
  className="page-wrapper"
>
  <div
    className="panel-group accordion"
  >
    <ApplicationPreferencesPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      formState={Object {}}
      isSaveButton={true}
      isShowControlPanel={true}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onSaveSettings={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
      patientsInfo={
        Object {
          "browserTitle": "PulseTile",
          "logoB64": "data:image/gif;base64,R0lGODlhDwAPAKECAAAAzMzM/////wAAACwAAAAADwAPAAACIISPeQHsrZ5ModrLlN48CXF8m2iQ3YmmKqVlRtW4MLwWACH+H09wdGltaXplZCBieSBVbGVhZCBTbWFydFNhdmVyIQAAOw==",
          "themeColor": "green",
          "title": "ripple",
        }
      }
      theme={
        Object {
          "baseColor": "#0D672F",
          "name": "Green Theme",
        }
      }
    />
    <PersonalInformationPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      isSaveButton={false}
      isShowControlPanel={true}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
      user={
        Object {
          "email": "bob.smith@gmail.com",
          "family_name": "Smith",
          "given_name": "Bob",
          "role": "IDCR",
          "roles": Array [
            "IDCR",
          ],
          "sub": "28AD8576-1948-4C84-8B5E-55FB7EE027CE",
          "tenant": null,
        }
      }
    />
    <ContactInformationPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      isSaveButton={false}
      isShowControlPanel={true}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
      user={
        Object {
          "email": "bob.smith@gmail.com",
          "family_name": "Smith",
          "given_name": "Bob",
          "role": "IDCR",
          "roles": Array [
            "IDCR",
          ],
          "sub": "28AD8576-1948-4C84-8B5E-55FB7EE027CE",
          "tenant": null,
        }
      }
    />
    <ChangeHistoryPanel
      editedPanel={Object {}}
      expandedPanel="applicationPreferences"
      isSaveButton={false}
      isShowControlPanel={false}
      onCancel={[Function]}
      onEdit={[Function]}
      onExpand={[Function]}
      onShow={[Function]}
      openedPanel="applicationPreferences"
    />
  </div>
</section>
`;
